File: Languages/Python/bee_movie.py
print("Hello, World!")

import os
from collections import Counter

def analyze_bee_movie():
    script_path = "../beemoviescript.txt"
    
    if not os.path.exists(script_path):
        print(f"File not found: {script_path}")
        return
    
    # Read file line by line and print each line
    print("Bee Movie Script:")
    print("-" * 50)
    
    with open(script_path, 'r', encoding='utf-8') as f:
        lines = f.readlines()
        for line in lines:
            print(line.rstrip())
    
    print("-" * 50)
    print("Analysis complete.")
    
    # Count letter frequencies
    all_text = ''.join(lines)
    letters = [char.lower() for char in all_text if char.isalpha()]
    
    if letters:
        letter_counts = Counter(letters)
        top_3 = letter_counts.most_common(3)
        
        print("\nTop 3 most commonly used letters:")
        for i, (letter, count) in enumerate(top_3, 1):
            print(f"{i}. '{letter}': {count} times")
    else:
        print("No letters found in the script.")

if __name__ == "__main__":
    analyze_bee_movie()

File: Languages/R/bee_movie.r
# Hello, World!
cat("Hello, World!\n")

analyze_bee_movie <- function() {
  script_path <- "../beemoviescript.txt"
  
  if (!file.exists(script_path)) {
    cat("File not found:", script_path, "\n")
    return()
  }
  
  cat("Bee Movie Script:\n")
  cat("--------------------------------------------------\n")
  
  # Read file line by line and print each line
  lines <- readLines(script_path)
  
  # Print each line
  for (line in lines) {
    cat(line, "\n")
  }
  
  cat("--------------------------------------------------\n")
  cat("Analysis complete.\n")
  
  # Count letters
  all_text <- paste(lines, collapse = "")
  letters <- strsplit(all_text, NULL)[[1]]
  letters <- letters[grepl("[a-zA-Z]", letters)]
  lower_letters <- tolower(letters)
  
  if (length(letters) > 0) {
    letter_counts <- table(lower_letters)
    sorted_counts <- sort(letter_counts, decreasing = TRUE)
    top_3 <- head(sorted_counts, 3)
    
    cat("\nTop 3 most commonly used letters:\n")
    for (i in 1:length(top_3)) {
      letter <- names(top_3)[i]
      count <- top_3[i]
      cat(i, ". '", letter, "': ", count, " times\n", sep = "")
    }
  } else {
    cat("No letters found in the script.\n")
  }
}

analyze_bee_movie()
